packages/fetchai/connections/soef/connection.py

```diff
@@ -1071,14 +1073,23 @@ async def _check_server_reachable(self) -> None:
             )
         except asyncio.TimeoutError:
             raise SOEFNetworkConnectionError(
-                f"Server can not be reached within timeout = {self.connection_check_timeout}!"
+                f"Server can not be reached within timeout={self.connection_check_timeout}!"
             )
 
     async def connect(self) -> None:
         """Connect channel set queues and executor pool."""
         self._loop = asyncio.get_event_loop()
 
-        await self._check_server_reachable()
+        reachable_check_count = 0
+        while reachable_check_count < self.connection_check_max_retries:
+            reachable_check_count += 1
+            try:
+                await self._check_server_reachable()
+                reachable_check_count = self.connection_check_max_retries
+            except Exception as e:  # pragma: nocover
+                if reachable_check_count == self.connection_check_max_retries:
+                    raise e
+                self.logger.debug(f"Exception during SOEF reachability check: {e}.")
```

Comment:
consider handling `SOEFNetworkConnectionError` explicitly, and print an appropriate error message.

Reply:
Something like:
```suggestion
            except SOEFNetworkConnectionError as e:
                if reachable_check_count == self.connection_check_max_retries:
                    self.logger.error("Max retries")
                    raise #TODO MaxRetriesError
                self.logger.debug(f"Exception during SOEF reachability check: {e}.")
```

Reply:
The problem is where missing multiple connection errors (not a full enumeration is done). I'd rather capture any exception until this is properly broken down.